src/main/java/com/lishid/openinv/internal/SpecialEnderChest.java

```diff
@@ -82,6 +81,7 @@ public void playerOffline() {
     public void onClose(CraftHumanEntity who) {
         super.onClose(who);
         saveOnExit();
+        OpenInv.enderChests.remove(owner.getUniqueId());
```

Comment:
Where the code was saveAndExit was only causing a save when the player goes offline (as the ticket describe). The following would happen:

- Operator opens player inventory. Player online.
- Operator closes inventory. Player online. Save is skipped since playerOnline boolean is still set to true. Inventory remains in OpenInv.enderChests.
- Players quits. And the order I mentioned above was firing:
- - PlayerQuitEvent (Monitor priority) -> OpenInv save inventory (the correct data)
- - Player saves as normal (the old data) resetting the players inventory and position